src/eval_cache.rs

```diff
@@ -108,21 +108,19 @@ impl CacheManifest {
 
     /// Checks and inserts the formatter info into the cache.
     /// If the formatter info has changed, invalidate all the old paths.
-    #[must_use]
-    pub fn update_formatters(self, formatters: BTreeMap<FormatterName, Formatter>) -> Self {
-        let mut new_formatters = BTreeMap::new();
-        let mut new_paths = self.matches.clone();
+    pub fn update_formatters(&mut self, formatters: BTreeMap<FormatterName, Formatter>) {
+        let mut old_formatters = std::mem::replace(&mut self.formatters, BTreeMap::new());
         for (name, fmt) in formatters {
-            match load_formatter_info(&fmt) {
+            match load_formatter_info(fmt) {
```

Comment:
Because we already have a whole cloned "copy" of `formatters` (and therefore `fmt`) in this instance, we might as well just consume them to prevent some clones below